src/redis-cli.c

```diff
@@ -382,6 +391,43 @@ void dictListDestructor(dict *d, void *val)
     listRelease((list*)val);
 }
 
+/* Erase the lines before printing, and returns the number of lines printed */
+int cleanPrintln(char *string) {
+    int line_length;
+    char *position = strchr(string, '\n');
+
+    /* Clear the line if in TTY */
+    if (IS_TTY_OR_FAKETTY()) {
+        printf("\033[2K\r");
+    }
+
+    if (position) {
+        line_length = (int)(position - string);
+        printf("%.*s\n", line_length, string);
+        return cleanPrintln(++position) + 1;
+    } else {
+        printf("%s\n", string);
+        return 1;
+    }
+}
+
+/* Erase the lines before printing, and returns the number of lines printed */
+int cleanPrintfln(char *fmt, ...) {
+    va_list args;
+    char buf[1024]; /* limitation */
+    int char_count;
+
+    va_start(args, fmt);
+    char_count = vsnprintf(buf, sizeof(buf), fmt, args);
+    va_end(args);
+
+    if (char_count >= (int)sizeof(buf)) {
+        fprintf(stderr, "Warning: String was trimmed in cleanPrintln\n");
+    }
+
+    return cleanPrintln(buf);
+}
```

Comment:
what about eliminating `cleanPrintln()` which is only used by `cleanPrintfln()`.
```suggestion
/* Erase the lines before printing, and returns the number of lines printed */
int cleanPrintfln(char *fmt, ...) {
    va_list args;
    char buf[1024]; /* limitation */
    int char_count, line_count = 0;

    /* Clear the line if in TTY */
    if (IS_TTY_OR_FAKETTY()) {
        printf("\033[2K\r");
    }

    va_start(args, fmt);
    char_count = vsnprintf(buf, sizeof(buf), fmt, args);
    va_end(args);

    if (char_count >= (int)sizeof(buf)) {
        fprintf(stderr, "Warning: String was trimmed in cleanPrintln\n");
    }

    char *position, *string = buf;
    while ((position = strchr(string, '\n')) != NULL) {
        int line_length = (int)(position - string);
        printf("%.*s\n", line_length, string);
        string = position + 1;
        line_count++;
    }

    printf("%s\n", string);
    return line_count + 1;
}
```

Reply:
Looks better and easier to read!
Without the recursion we can merge both functions.